File: fe/src/features/TestOrder/types/TestComments.ts
// src/features/TestOrder/types/TestComments.ts

export type TestCommentDTO = {
  id: string;
  testOrderId: string;
  testResultId: string;
  userId: string;
  content: string;
  createdAt: string;
  updatedAt?: string | null;
};

export type AdminTestCommentDTO = {
  id: string;
  testOrderId: string;
  testResultId: string;
  userId: string;
  content: string;
  createdAt: string;
  updatedAt?: string | null;
  isDeleted: boolean;
};

File: be/test-order-service/src/main/java/org/overcode250204/testorderservice/models/entites/TestComments.java
package org.overcode250204.testorderservice.models.entites;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "test_comments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TestComments {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "test_order_id", nullable = false)
    private TestOrders testOrder;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "test_result_id")
    private TestResults testResult;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "content", nullable = false, columnDefinition = "TEXT")
    private String content;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "is_deleted")
    private Boolean isDeleted = false;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = createdAt;
    }
}